### PR TITLE
verify: Added safe delete.

### PR DESCRIPTION
**Pull request by Bplotka**
Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: pkg/objstore/client/factory.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+	"cloud.google.com/go/storage"
+	"github.com/improbable-eng/thanos/pkg/objstore"
+	"github.com/improbable-eng/thanos/pkg/objstore/gcs"
+	"github.com/improbable-eng/thanos/pkg/objstore/s3"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var NotFoundError = errors.New("no valid GCS or S3 configuration supplied")
```
Comment (fabxc) [inline pkg/objstore/client/factory.go]: `ErrNotFound`

File: cmd/thanos/bucket.go [hunk 2]
```diff
@@ -49,18 +47,38 @@ func registerBucket(m map[string]setupFunc, app *kingpin.Application, name strin
 	verify := cmd.Command("verify", "verify all blocks in the bucket against specified issues")
 	verifyRepair := verify.Flag("repair", "attempt to repair blocks for which issues were detected").
 		Short('r').Default("false").Bool()
+	// NOTE(bplotka): Currently we support backup buckets only in the same project.
+	verifyBackupGCSBucket := cmd.Flag("gcs-backup-bucket", "Google Cloud Storage bucket name to backup blocks on repair operations.").
+		PlaceHolder("<bucket>").String()
+	verifyBackupS3Bucket := cmd.Flag("s3-backup-bucket", "S3 bucket name to backup blocks on repair operations.").
+		PlaceHolder("<bucket>").String()
```
Comment (fabxc) [inline cmd/thanos/bucket.go]: How long until someone requests to provide separate credentials for these :P?

Comment (Bplotka) [reply cmd/thanos/bucket.go]: I bet -> on next compactor bug (if any)
Hopefully, at that point we will have shiny https://github.com/improbable-eng/thanos/issues/209

(:

**Comment by fabxc** [comment]
I'd generally hope this isn't something we need regular production users to use. 👍

**Comment by Bplotka** [comment]
of course (:

**Pull request merged**